File: 0035-search-insert-position/0035-search-insert-position.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        if (target>nums[nums.size()-1]){
            return nums.size();
        }
        if (target<=nums[0]){
            return 0;
        }
        for (int i=1;i<=nums.size();i++){
            if (target>nums[i-1] and target<=nums[i]){
                return i;
            }
        }
        return nums.size();
    }
};

File: 0035-search-insert-position/0035-search-insert-position.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        if target<nums[0]:
            return 0
        if target>nums[len(nums)-1]:
            return len(nums)
        l=0
        r=len(nums)-1
        while(l<=r):
            mid=l+(r-l)//2
            if nums[mid]==target or (nums[mid-1]<target and nums[mid]>target) :
                return mid
            
            if nums[mid]<target:
                l=mid+1
                continue
            else:
                r=mid-1
                continue